src/Compilers/Core/Portable/Emit/EditAndContinue/DeltaMetadataWriter.cs

```diff
@@ -784,6 +812,266 @@ private void PopulateEncLogTableParameters()
+                var emittedParent = emittedAttributeParents[emittedIndex];
+                var emittedCodedIndex = CodedIndex.HasCustomAttribute(emittedParent);
+
+                if (parentCodedIndex > emittedCodedIndex)
+                {
+                    // If the existing data is ahead of the emitted data then it means the current emitted attribute
+                    // must be for something that wasn't in the original compilation so we'll deal with it later.
+                    skippedParents.Add(emittedParent);
+
+                    // Move our log row index on enough to skip over the remaining attributes for this parent, which will
+                    // be filled in later when we process the deltas
+                    encLogRowId += _customAttributeParentCounts[emittedParent];
+
+                    // We're skipping the emitted attribute, so move that index on, but we want to re-check the
+                    // existing attribute in case it matches our next emitted attribute
+                    emittedIndex++;
+                }
+                else if (parent == emittedParent)
+                {
+                    // If we found the attribute we want then add an EncLog table record for it in the right position
+                    // If the right position doesn't exist, then that means we found a parent for which an attribute has
+                    // been deleted
+                    if (encLogRows.Count <= encLogRowId)
+                    {
+                        // Expand our enc log table to accommodate
+                        encLogRows.Add(0);
+                        var ca = metadataReader.GetCustomAttribute(customAttributeHandle);
+                        AddDeletedCustomAttribute(ca.Parent.Kind, ca.Constructor, ca.Value);
+                    }
+                    encLogRows[encLogRowId] = rowId;
+
+                    // We matched, so decrement the count of attributes emitted for this parent
+                    _customAttributeParentCounts[emittedParent]--;
+                    // If there are no more attributes for this parent to emit, then move on to the next
+                    if (_customAttributeParentCounts[emittedParent] <= 0)
+                    {
+                        emittedIndex++;
+                    }
+                    encLogRowId++;
+                    rowId++;
+                    hasData = enumerator.MoveNext();
+                }
+                else
+                {
+                    // No match, but ordering is how we want it, so we're still looking for our current emitted parent
+                    rowId++;
+                    hasData = enumerator.MoveNext();
+                }
+
+                // If we've looked for everything we emitted we can stop looking in existing metadata.
+                // We'll deal with anything we've skipped later.
+                if (emittedIndex == emittedAttributeParents.Length)
```

Comment:
>emittedIndex == emittedAttributeParents.Length [](http://example.com/codeflow?start=20&length=46)

Consider combining the last statement of the loop with the condition into:
`hasData && emittedIndex < emittedAttributeParents.Length`